lib/libc/minimal/source/stdout/stdout_console.c

```diff
@@ -50,7 +50,7 @@ int fputs(const char *_MLIBC_RESTRICT string, FILE *_MLIBC_RESTRICT stream)
 
 	ret = fwrite(string, len, 1, stream);
 
-	return len == ret ? 0 : EOF;
+	return ret > 0 ? 0 : EOF;
```

Comment:
Shouldn't this be `ret >= 0`?

Reply:
`fwrite` returns value which is number of members successfully written. It is equal to `nitems` in case of success. In this place we call `fwrite` with `nitems` = 1, so `fwrite` will return 1 in case of success and 0 in case of failure.

As we are checking against `fwrite` success we should use `ret > 0` condition here.

Reply:
OK, I see in fwrite(3):

> If an error occurs,  or  the  end  of  the  file  is reached, the return value is a short item count (or zero).

I was thinking of the legitimate (if unoptimized) use of writing zero items, but if we want to consider that an error, I guess it's unlikely to happen.

Reply:
Actually, looking at this closely, the correct change is either:

    return (ret == len) ? 0 : EOF

or to change the call:

    ret = fwrite(string, 1, len, stream);

IOW the call and the result check are just wrong.

Reply:
I guess
```
ret = fwrite(string, 1, len, stream);
```
would be good candidate here. I'll update PR.

Reply:
Done!